ui/src/main/webapp/src/app/reports/migration-issues/migration-issues.service.ts

```diff
@@ -20,10 +23,79 @@ export class MigrationIssuesService extends AbstractService {
             .catch(this.handleError);
     }
 
+    @Cached({section: 'migrationIssues', immutable: true})
+    getDataFromServer(executionId: number) {
+        let url = MigrationIssuesService.NEW_AGGREGATED_ISSUES_URL.replace('{executionId}', executionId.toString());
+
+        return this._http.get(url)
+            .map(res => res.json())
+            .catch(this.handleError);
+    }
+
+    getNewAggregatedIssues(executionId: number, filter?: ReportFilter): Observable<Dictionary<ProblemSummary[]>> {
+        return this.getDataFromServer(executionId)
+            .map((serverResponse: MigrationIssuesResponse[]) => {
+                let result: Dictionary<ProblemSummary[]> = {};
+
+                let data  = serverResponse;
+
+
+                if (filter) {
+                    data = serverResponse.filter(item => {
+                        return filter.selectedApplications.some(filterApp => {
+                            let rootFileName =  (<any>item.projectModel).vertices_out.rootFileModel.vertices[0].fileName;
+
+                            if ((<FilterApplication>filterApp).fileName) {
+                                return rootFileName === (<FilterApplication>filterApp).fileName;
+                            } else {
+                                return rootFileName === (<Partial<RegisteredApplication>>filterApp).inputFilename;
+                            }
+                        });
+                    });
+                }
```

Comment:
I am still a little concerned about scalability with the very large projects, especially if this same approach were used on the application details page. I could be mistaken, though. It does seem like this will improve performance in some cases.

I guess if you do the quick workaround approach to getting filtering working for this sprint then we can compare the two approaches for the release?